### PR TITLE
rocAL: Classification Training Related changes

### PR DESCRIPTION
**Pull request by swetha097**
1. Rng changes similar to DALI
2. Added area_factor, aspect_ratio, num_attemps as params for partial decoders
3. Centre Crop Fix
4. Minor CMN fix
5. Wrapped long lines of code in python files
6. Set the seed for the entire pipeline once in Pipeline.py
7. Added decode_size_policy, max_decoded_width, max_decoded_height that can be user passable from rocAL python API
8. Remove py::args from the decoders in rocal_pybind.cpp file

### CODE REVIEW

**Comment by LakshmiKumar23** [comment]
@swetha097 can you sync to TOT?

**Comment by rrawther** [comment]
@swetha097: can you merge the tot changes for this PR

**Review by rrawther — APPROVED** [review]
@LakshmiKumar23 to test all unit-tests and training for rocAL

**Comment by LakshmiKumar23** [comment]
@swetha097 Training runs without any errors. 
Test cases - caffe classification error (similarly for all test cases)
```
>>> test case 0
>>> Running on GPU ,  Color
OK: OpenVX using GPU device - 0:  [gfx90a:sramecc+:xnack-] with 110 CUs on PCI bus c1:00.0
OK: loaded 81 kernels from libvx_rpp.so
>>>>>>> Running CAFFE CLASSIFICATION READER
terminate called after throwing an instance of 'RocalException'
  what():   { read_lmdb_record } LMDB error, /dockerx/MIVisionX/rocAL/rocAL/source/meta_data/caffe_meta_data_reader.cpp:134 mdb_env_open(_mdb_env, _path.c_str(), MDB_RDONLY, 0664):No such fil
e or directory
Aborted (core dumped)
```

File: rocAL/rocAL/include/api/rocal_api_data_loaders.h [hunk 1]
```diff
@@ -392,27 +389,27 @@ extern "C"  RocalImage  ROCAL_API_CALL rocalMXNetRecordSourceSingleShard(RocalCo
...
+                                                        std::vector<float>& aspect_ratio,
+                                                        unsigned num_attempts,
                                                         bool shuffle = false,
                                                         bool loop = false,
-                                                        RocalImageSizeEvaluationPolicy decode_size_policy = ROCAL_USE_MAX_SIZE,
-                                                        unsigned max_width = 0, unsigned max_height = 0,
-                                                        RocalFloatParam area_factor = NULL, RocalFloatParam aspect_ratio = NULL,
-                                                        RocalFloatParam y_drift_factor = NULL, RocalFloatParam x_drift_factor = NULL);
+                                                        RocalImageSizeEvaluationPolicy decode_size_policy = ROCAL_USE_MOST_FREQUENT_SIZE,
+                                                        unsigned max_width = 0, unsigned max_height = 0);
```
[the remark this answers is not in the document]
Comment (swetha097) [reply rocAL/rocAL/include/api/rocal_api_data_loaders.h:412]: We used MOST_FREQUENT_SIZE as the "default policy" keeping in mind the memory allocation for the majority of the images in the entire dataset is not huge when user does not specify any policy.
In training's however, we have been using USER_GIVEN_SIZE_RESTRICTED with the given max_decoded_width or height specified by the user.

**Comment by swetha097** [comment]
> @swetha097 Training runs without any errors. Test cases - caffe classification error (similarly for all test cases)
> 
> ```
> >>> test case 0
> >>> Running on GPU ,  Color
> OK: OpenVX using GPU device - 0:  [gfx90a:sramecc+:xnack-] with 110 CUs on PCI bus c1:00.0
> OK: loaded 81 kernels from libvx_rpp.so
> >>>>>>> Running CAFFE CLASSIFICATION READER
> terminate called after throwing an instance of 'RocalException'
>   what():   { read_lmdb_record } LMDB error, /dockerx/MIVisionX/rocAL/rocAL/source/meta_data/caffe_meta_data_reader.cpp:134 mdb_env_open(_mdb_env, _path.c_str(), MDB_RDONLY, 0664):No such fil
> e or directory
> Aborted (core dumped)
> ```

@LakshmiKumar23 
I tested this out, but did not seem to get this error . 
However, looks like the folder passed to the rocAL unit test for caffe / caffe2 readers - the image folder path has to be passed one level down, if not this issue is seen.

MIVisionX-data/rocal_data/caffe/classification/ilsvrc12_train_lmdb - Works

**Pull request merged**